### PR TITLE
Revise python test and dependencies to sample applications

### PR DESCRIPTION
**Pull request by piponazo**
As discussed today in our chat, there was some confusion about how to run the tests. It turns out we still had in the `test` folder some old Makefiles and bash scripts that were used to run the system-tests over the **exiv2** application and some of the sample applications.

In this PR I did 2 things:
- Remove the deprecated bash files & Makefile. All these tests were already ported to the python test suite.
- Move some tests from `tests/bugfixes` to `tests\bash_tests`. Even though the files moved were indeed added while fixing some bugs, they depend on sample applications. We can discuss if that `bash_tests` folders is the right place for these files or whether we should create another category (something like `tests\sample_tests`). But the main idea is to keep the `bugfixes` folder just to test the exiv2 library and application.

### CODE REVIEW

**Review by kevinbackhouse — DISMISSED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -110,25 +110,28 @@ if( EXIV2_BUILD_SAMPLES )
...
+    )
+    add_test(NAME tiffTests
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/tests
+        COMMAND cmake -E env EXIV2_BINDIR=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} ${Python3_EXECUTABLE} runner.py --verbose tiff_test
+    )
+    add_test(NAME versionTests
+        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/tests
+        COMMAND cmake -E env EXIV2_BINDIR=${CMAKE_RUNTIME_OUTPUT_DIRECTORY} ${Python3_EXECUTABLE} runner.py --verbose bash_tests/version_test.py 
+    )
+endif()
```
Comment (hassec) [inline CMakeLists.txt:133]: Let's add a warning message in an `else` block here just so we don't silently not add any tests.

**Review by hassec — DISMISSED** [review]
LGTM :+1:

**Review by hassec — APPROVED** [review]

**Comment by clanmills** [comment]
That's awesome work, @piponazo.  README.md and CONTRIBUTING.md still discuss `make tests`.  I'm bringing this up because that's why @mallman opened #2082.  I discussed this with him and he closed the issue report.  However the documents have not been reviewed and corrected.

```
574 rmills@rmillsmm-local:~/gnu/github/exiv2/main $ find . -name "*.md" | xargs grep 'make ' | grep -v 'cmake '
./exiv2.md:requirements of the target format. Some tags copied like this may not make 
./contrib/vs2019/README.md:consistent with all the choices that I make for the application (like
./contrib/vs2019/README.md:project. This make Visual Studio generate the correct dependency
./GIT_GUIDELINES.md:Commits should be atomic, i.e. they should make one self-contained
./GIT_GUIDELINES.md:- Changes made in different source which do not make sense without each other
./README.md:$ sudo make uninstall
./README.md:$ make                                             # build the code
./README.md:$ make CXX_FLAGS=-DEXIV2_DEBUG_MESSAGES
./README.md:$ sudo make install
./README.md:$ make tests
./README.md:| VERBOSE            | _**not set**_              | Makefile platforms | Instructs make to report its actions |
./README.md:for i in base-devel git coreutils dos2unix tar diffutils make                     \
./CONTRIBUTING.md:6. Now, make your change(s), add tests for your changes, and commit each change:
./CONTRIBUTING.md:        $ make tests         # Integration tests
./doc/readme-pvs-studio.md:The PVS username & key are configured as secrets in the security settings of the project. Then we make use of such
./fuzz/README.md:make -j $(nproc)
575 rmills@rmillsmm-local:~/gnu/github/exiv2/main $ 
```

**Comment by piponazo** [comment]
Thanks @clanmills for letting me know. I'll go through the different documents and try to fix all these occurrences

**Comment by clanmills** [comment]
Good stuff, @piponazo.  @kmilos and I agreed that all documentation should always discuss build and testing using cmake and conan.  We should avoid discussing "legacy" tools such as make.  So the preferred way to build is:

```bash
$ cd <exiv2dir>
$ mkdir build 
$ cd build
$ cmake ..
$ cmake --build .
$ ctest
$ sudo make install
```

I think Milos may prefer to build in a totally different location.  I can't remember exactly how to run ctest in a totally different directory.  The preferred build/test command are something like:

```bash
$ cd <exiv2dir>
$ mkdir /tmp/exiv2-build
$ cmake . --B /tmp/exiv2-build   [-DCMAKE_BUILD_TYPE=RELEASE]
$ cmake --build /tmp/exiv2-build [--target target] [--config Release]
$ ctest --build-run-dir /tmp/exiv2-build  # This is wrong.  I don't know how to do this!
$ sudo cmake --build /tmp/exiv2-build --target install
```
@kmilos.  Can you comment please?

**Comment by kmilos** [comment]
I don't think I was doing anything different to what CI scripts do, and there we run something like `ctest --test-dir /tmp/exiv2-build`

Btw, you no longer need `mkdir /tmp/exiv2-build`, as `cmake . --B /tmp/exiv2-build` will create it for you. You might want to `rm -rf` it before or after the whole process though to have a clean slate...

**Comment by piponazo** [comment]
@clanmills You can find some examples of how to build from other directories in the CI pipelines under `exiv2/.github/workflows`.

In my opinion, we should prefer to document things as in your first code snippet. It makes the documentation less verbose and I think it is the approach most of users will use when working locally.

The second approach is specially useful for CI pipelines. One can sacrifice there some more characters to be more explicit about what are the source_code & build directories, without needing to move across directories.

**Comment by clanmills** [comment]
@kmilos.  Thanks for your input.  That's great.

And thank you to @piponazo for his efforts on this issue concerning python.  I think we should keep the names "bashTests" and "bugfixTests" because change usually leads to confusion!

I'll deal with CONTRIBUTING.md and README.md at the weekend to service: #2082.  When I open the PR, I'll invite you both to review.

**Pull request merged**